cognite_toolkit/_cdf_tk/commands/_profile.py

```diff
@@ -185,6 +187,251 @@ def _as_cell(self, value: PendingCellValue) -> str | Spinner:
+        elif col == self.Columns.Transformations:
+            if isinstance(result, list) and len(result) > 0 and isinstance(result[0], Transformation):
+                return f"{result[0].name} ({result[0].external_id})"
+            return None
+        elif col == self.Columns.RowCount:
+            if isinstance(result, RawProfileResults):
+                return result.row_count
+            return None
+        raise ValueError(f"unexpected result type {type(result)} for row {row} and column {col}.")
+
+    def update_table(
+        self,
+        current_table: PendingTable,
+        result: object,
+        row: str,
+        col: str,
+    ) -> PendingTable:
+        handlers: Mapping[str, Callable[[PendingTable, object, str], PendingTable]] = {
+            self.Columns.Count: self._update_count,
+            self.Columns.DataSets: self._update_datasets,
+            self.Columns.DataSetCount: self._update_dataset_count,
+            self.Columns.Transformations: self._update_transformations,
+            self.Columns.RowCount: self._update_row_count,
+        }
+        handler = handlers.get(col)
+        if handler:
+            return handler(current_table, result, row)
+        return current_table
+
+    def _update_count(
+        self,
+        current_table: PendingTable,
+        result: object,
+        row: str,
+    ) -> PendingTable:
+        new_table: PendingTable = {}
+        for (r, c), value in current_table.items():
+            if r == row and c == self.Columns.DataSets:
+                new_table[(r, c)] = WaitingAPICall
+            else:
+                new_table[(r, c)] = value
+        return new_table
+
+    def _update_datasets(
+        self,
+        current_table: PendingTable,
+        result: object,
+        row: str,
+    ) -> PendingTable:
+        if not (isinstance(result, list) and len(result) > 0 and all(isinstance(item, str) for item in result)):
+            return current_table
+        new_table: PendingTable = {}
+        for (r, c), value in current_table.items():
+            if r != row:
+                new_table[(r, c)] = value
+                continue
+            for data_set in result:
+                new_index = f"{r}{self._index_split}{data_set}"
+                if c == self.Columns.DataSetCount:
+                    new_table[(new_index, c)] = WaitingAPICall
+                elif c == self.Columns.DataSets:
+                    new_table[(new_index, c)] = data_set
+                else:
+                    new_table[(new_index, c)] = value
+        return new_table
+
+    def _update_dataset_count(
+        self,
+        current_table: PendingTable,
+        result: object,
+        row: str,
+    ) -> PendingTable:
+        if not isinstance(result, int):
+            return current_table
+        new_table: PendingTable = {}
+        for (r, c), value in current_table.items():
```

Comment:
Nitpick:
If possible, can you avoid single char variable names? Affects code readability, in general.